File: library/src/main/java/me/proxer/library/api/info/ModifyUserInfoEndpoint.java
package me.proxer.library.api.info;

import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;

/**
 * Endpoint for adding an entry to the notes, favorites or finished entries.
 *
 * @author Ruben Gees
 */
public final class ModifyUserInfoEndpoint implements Endpoint<Void> {

    private final InternalApi internalApi;

    private final String id;
    private final UserInfoType type;

    ModifyUserInfoEndpoint(final InternalApi internalApi, final String id, final UserInfoType type) {
        this.internalApi = internalApi;
        this.id = id;
        this.type = type;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<Void> build() {
        return internalApi.modifyUserInfo(id, type);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/info/ModifyUserInfoEndpoint.kt
package me.proxer.library.api.info

import me.proxer.library.api.Endpoint
import me.proxer.library.api.ProxerCall

/**
 * Endpoint for adding an entry to the notes, favorites or finished entries.
 *
 * @author Ruben Gees
 */
class ModifyUserInfoEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val id: String,
    private val type: UserInfoType
) : Endpoint<Unit> {

    override fun build(): ProxerCall<Unit> {
        return internalApi.modifyUserInfo(id, type)
    }
}
